include/oneapi/dpl/pstl/algorithm_ranges_impl.h

```diff
@@ -0,0 +1,103 @@
+// -*- C++ -*-
```

Comment:
As far as I understand we are going to release just hetero version of reduce_by_segment, right?
In that case, I would suggest creating new PR for hetero version.. (by keeping it as a proto)..
Or, create new PR for host version prototype and remove relevant code from this PR.

Reply:
@rarutyun, what's your opinion?  The focus has been on the hetero version, but I don't know why we wouldn't release the host version as well.

Reply:
working through CI errors it seems that just the hetero version is ready at this time.  I'll move the host version prototype to a new PR.